sdk-extensions/tracing-incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/trace/WorkerExporter.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.trace;
+
+import io.opentelemetry.api.metrics.BoundLongCounter;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class WorkerExporter {
+
+  private final SpanExporter spanExporter;
+  private final ScheduledExecutorService executorService;
+  private final Logger logger;
+  private final long exporterTimeoutNanos;
+  private final BoundLongCounter exportedSpans;
+  private final AtomicReference<CompletableResultCode> flushRequested;
+  private final int maxExportBatchSize;
+
+  WorkerExporter(
```

Comment:
Seems weird to have a public class with all public methods with a non-public constructor. Does this class need to be public right now?

Reply:
nope, it can be in package scope